azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsProperties.java

```diff
@@ -91,7 +93,17 @@ public void setEnabled(boolean enabled) {
   }
 
   public String getInstrumentationKey() {
-    return instrumentationKey;
+    // First try getting ikey from application.properties
```

Comment:
I don't understand this code. If the key is present in `application.properties` or as a System property or OS Env variables, it will be injected. Doing this manually either means you're post-processing the key, or something else is wrong.

Reply:
As I mentioned in the comment above instrumentation key can come from any of the three locations and therefore I had the need to perform this logic. Let me know if you think there is a better way to do it.